novelist.py

```diff
@@ -0,0 +1,309 @@
+import json
+import logging
+import urllib
+from nose.tools import set_trace
+
+from core.config import Configuration
+from core.coverage import (
+    CoverageFailure,
+    CoverageProvider,
+)
+from core.metadata_layer import (
+    ContributorData,
+    IdentifierData,
+    LinkData,
+    MeasurementData,
+    Metadata,
+    SubjectData,
+)
+from core.model import (
+    DataSource,
+    Hyperlink,
+    Identifier,
+    Measurement,
+    Representation,
+    Subject,
+)
+from core.util import TitleProcessor
+
+class NoveListAPI(object):
+
+    log = logging.getLogger("NoveList API")
+    version = "2.2"
+
+    # While the NoveList API doesn't require parameters to be passed via URL,
+    # the Representation object needs a unique URL to return the proper data
+    # from the database.
+    QUERY_ENDPOINT = "http://novselect.ebscohost.com/Data/ContentByQuery?\
+            ISBN=%(ISBN)s&ClientIdentifier=%(ClientIdentifier)s&version=%(version)s"
+    MAX_REPRESENTATION_AGE = 6*30*24*60*60      # six months
+
+    @classmethod
+    def from_config(cls, _db):
+        config = Configuration.integration(Configuration.NOVELIST_INTEGRATION)
+        profile = config.get(Configuration.NOVELIST_PROFILE)
+        password = config.get(Configuration.NOVELIST_PASSWORD)
+        if not (profile and password):
+            raise ValueError("No NoveList client configured.")
+        return cls(_db, profile, password)
+
+    def __init__(self, _db, profile, password):
+        self._db = _db
+        self.profile = profile
+        self.password = password
+        self.source = DataSource.lookup(self._db, DataSource.NOVELIST)
```

Comment:
I recently decided that storing DataSources like this is bad practice because if this object ever gets used in a multithreaded web app, it'll break pretty quickly. I know that this object isn't currently used in a web app, but it might easily be in a metadata wrangler admin app. So I'd recommend (at least for now) doing what I did in core/nyt.py, something like this:

```
    @property
    def source(self):
        return DataSource.lookup(_db, DataSource.NOVELIST)
```